### DISCUSSION TITLE
rsync ~/.ssh/config autocompletion doesn't work

### DISCUSSION THREAD

**Discussion by zepx** [ohmyzsh/ohmyzsh#11971] [Help]
Hi,

I am currently running on ArchLinux as well as on Ubuntu 16.06. For commands `scp` and `ssh`. For both of these distros, oh-my-zsh's autocomplete works fine and I could easily pull custom host names configured in `~/.ssh/config` and '~/.ssh/config.d/home.config` (on ArchLinux).

However, specifically for `rsync` I could not get this to work. To use host names configured in `~/.ssh/config`, I'm required to switch to `scp` or `ssh` first to utilize the autocomplete feature.

**Use Case**

`~/.ssh/config`:
```
Include config.d/*
```

`~/.ssh/config/home.config`:
```
Host home_server_vpn*
    Hostname 10.7.0.6
    DynamicForward 3310
    Host home_server_vpn_choong

Host home_*_choong
    User choong

Host *
    ServerAliveInterval 30
    ServerAliveCountMax 3
    ForwardAgent yes
    IdentityFile ~/.ssh/id_rsa.sec
```

1. To copy a file from a remote server to local server using `rsync`,
`$ rsync home<tab>`
does not give any autocompletion

2. However,
`$ scp home<tab>` works perfectly fine and would result  in
`$ scp home_server_vpn_choong`

**Comment by brandonStell** [comment]
@zepx have you found a solution yet? This is not working for me either.

**Comment by zepx** [comment]
Unfortunately no. I did not invest much time figuring this out. I sometimes use `ssh` or `scp` for completion then change the command to `rsync`. It's relatively inconvenient.

**Comment by stefabat** [comment]
I believe this is unrelated to oh-my-zsh because I experience the same on my machine as well (Arch Linux), but I am not using oh-my-zsh; I use plain zsh and antigen (without loading oh-my-zsh plugins).
So far, I have not found a solution yet.

**Comment by polaroidkidd** [comment]
this is afflicting me aswell unfortunately. completion of path on remotes work when using `rsync -raP my-host:` but the hostname in the `~/.ssh/config` file is not autocompleted as it is with `ssh` or `scp`

Unfortunately I haven't the foggiest where the hick up is :/


for what it's worth, here's my config

```

# Add wisely, as too many plugins slow down shell startup.
plugins=(
  git
  ssh-agent
  zsh-syntax-highlighting
  zsh-autosuggestions
 )
zstyle ':completion:*:(rsync|ssh|scp|ftp|sftp):*' users $users

zstyle :omz:plugins:ssh-agent agent-forwarding on
zstyle :omz:plugins:ssh-agent identities me@my-host
zstyle :omz:plugins:ssh-agent lifetime

source $ZSH/oh-my-zsh.sh

fpath=(~/.zsh/completion $fpath)
autoload -Uz compinit && compinit -i

...
```

**Comment by rklasen** [comment]
Same here, plain zsh without plugins on Arch. `ssh` host completion from `.ssh/config` works, but doesn't for `rsync`. Very interested if anyone finds a solution.

**Comment by simoneruffini** [comment]
Could this help https://stackoverflow.com/questions/54309712/zsh-doesnt-autocomplete-correctly-my-ssh-command ?
I don't know how this works, my only complaint is that if you have aliases on rsync/ssh/scp you need to put them too in-between parenthesis.
